lib/pat.c

```diff
@@ -1141,7 +1151,7 @@ key_put(grn_ctx *ctx, grn_pat *pat, const uint8_t *key, uint32_t len)
     }
     grn_memcpy(dest, key, len);
   }
-  pat->header->curr_key += len;
+  pat_increase_curr_key(pat, len);
```

Comment:
Since `header->curr_key += len` is only executed by this function, it seems unnecessary to create pat_increase_curr_key().